appd/README.md

```diff
@@ -0,0 +1,141 @@
+
+This means `nova` manages:
+
+- the CometBFT instance for all chains
+- the embedded binaries and their lifecycle
+- the version switch (upgrade process)
+- the connection between the standalone app, native app and CometBFT instance.
+
+By sitting between the standalone app and the CometBFT instance, `nova` is able to listen to an `AppVersion` change and trigger the upgrade process automatically.
+
+In order to limit any P2P disruption, like it happens today when no using `nova`; it uses one single CometBFT instance for all versions. This is possible because there has been no P2P and Block breaking changes in CometBFT.
+
+Nova supports ABCI 1.0 and ABCI 2.0, which means it can be used with any version of CometBFT, including the latest versions. This is done via the `ABCIClientVersion`.
+Nova has two remote clients, those remote clients interact with the App and CometBFT via gRPC:
+
+- `RemoteABCIClientV1`: This client is used for CometBFT v0.34 and below. It uses the `ABCIClientVersion` 1.
+- `RemoteABCIClientV2`: This client is used for CometBFT v0.38 and above. It uses the `ABCIClientVersion` 2.
+
+A chain defines a list of versions per `AppVersion` up until the last one:
+
+```go
+versions, err := abci.NewVersions(abci.Version{
+ Appd:        v3,
+ ABCIVersion: abci.ABCIClientVersion1,
+ AppVersion:  3,
+})
+```
+
+Multiple versions can be defined, allowing to sync from genesis with only one binary (for the node operators).
+
+Once the `AppVersion` changes, `nova` takes the best matching embedded binary (useful when syncing from genesis) or switch to the native binary if none matches (useful when upgrading).
+
+## Installation
+
+`Nova` integrates seemlessly in any Cosmos SDK chain. It simply replaces the `StartCommandHandler` of the chain:
+
+```diff
+server.AddCommandsWithStartCmdOptions(
+ rootCommand,
+ app.DefaultNodeHome,
+ NewApp,
+ appExporter,
+ server.StartCmdOptions{
+  AddFlags:            addStartFlags,
++ StartCommandHandler: nova.New(versions),
+ },
+)
+```
+
+Optionally add the passthrough mode command:
+
+```go
+rootCommand.AddCommand(
+ nova.NewPassthroughCmd(versions),
+)
+```
+
+## Customizing the embeedded binaries
+
+As `nova` manages the start and stop of embedded binaries and the connection between the standalone app <-> `nova` <-> CometBFT instance, it is important to ensure that the embedded binaries are properly started.
+
+Some chains may require custom flags while starting the binaries. This can be done when setting up the `abci.Version`:
+
+```go
+versions, err := abci.NewVersions(abci.Version{
+ Appd:        v3,
+ ABCIVersion: abci.ABCIClientVersion1,
+ AppVersion:  3,
+ StartArgs: []string{
+  "--grpc.enable=false",
+  "--api.enable=false",
+  "--api.swagger=false",
+  "--with-tendermint=false",
+  "--transport=grpc",
+ },
+})
+```
+
+The default list of flags, unless overridden is the following:
+
+```go
+"--grpc.enable=true",
```

Comment:
it might be nice to include a sample of `--grpc.address=0.0.0.0:9090` as I think it defaults to localhost binding sometimes which can cause problems.

Reply:
ok, i'll add in the example but not in the default, i think we actually reverted 0.0.0.0 to localhost from default config in the SDK for avoiding people leaking their node grpc servers.